sdk/test-utils/recorder/src/recorder.ts

```diff
@@ -136,6 +137,40 @@ export class Recorder {
     }
   }
 
+  /**
+   * revertRequestChanges reverts the request in record and playback modes back to the existing url.
+   * Works for both core-v1 and core-v2
+   *
+   * - WebResource -> core-v1
+   * - PipelineRequest -> core-v2
+   *
+   * Wrokflow:
+   *   1. recorderHttpPolicy calls this method after the request is made
+   *   2. "redirectRequest" method is called to update the request with the proxy-tool url
+   *   3. Request hits the proxy tool, proxy-tool hits the service and returns the response
+   *   4. Using `revertRequestChanges`, we revert the request back to the original url
+   */
+  private revertRequestChanges(request: WebResource | PipelineRequest): void {
+    logger.info(
+      `[Recorder#revertRequestChanges] "undo"s the URL changes made by the recorder to hit the test proxy after the response is received,`,
+      request
+    );
+    const headers = request.headers;
+    if (isHttpHeadersLike(headers)) {
+      // core-v1
+      headers.remove("x-recording-id");
+      headers.remove("x-recording-mode");
+    } else {
+      // core-v2
+      headers.delete("x-recording-id");
+      headers.delete("x-recording-mode");
+    }
+    request.url = request.url.replace(
+      Recorder.url,
+      request.headers.get("x-recording-upstream-base-uri") || "dummy"
```

Comment:
We found that our workaround had recently broken because "x-recording-upstream-base-uri" began storing only the original request host (rather than the full request URL). Wonder if that would cause problems here?

Reply:
It won't cause problems. It is true that the header only stores the endpoint and not the path in the URL.

Below is a workflow of an example URL.
```
// Initial URL
https://management.azure.com/subscriptions/faa080af-c1d8-40ad-9cce-e1a450ca5b57/providers/Microsoft.Network/locations/eastus/operations/633f1fe4-9e70-4519-98fb-8370f3ee0554?api-version=2022-05-01

// When it is sent to the test-proxy tool
http://localhost:5000/subscriptions/faa080af-c1d8-40ad-9cce-e1a450ca5b57/providers/Microsoft.Network/locations/eastus/operations/633f1fe4-9e70-4519-98fb-8370f3ee0554?api-version=2022-05-01

// With the new method from this PR, the url is changed back to the original
https://management.azure.com/subscriptions/faa080af-c1d8-40ad-9cce-e1a450ca5b57/providers/Microsoft.Network/locations/eastus/operations/633f1fe4-9e70-4519-98fb-8370f3ee0554?api-version=2022-05-01
```